docs/design/gatewayapi-translator.md

```diff
@@ -0,0 +1,242 @@
+# Gateway API Translator Design
+
+## Assumptions
+- initially target core conformance features only, to be followed by extended conformance features
+
+## Inputs and Outputs
```

Comment:
would be great if you can share the API or Input message/arg that the `Provider` will need to call/populate
cc @danehans who is working on it